src/lib.rs

```diff
@@ -819,8 +821,8 @@ impl Node {
 		self.channel_manager.get_our_node_id()
 	}
 
-	/// Returns our own listening address and port.
-	pub fn listening_address(&self) -> Option<String> {
+	/// Returns our own listening address.
+	pub fn listening_address(&self) -> Option<SocketAddr> {
 		self.config.listening_address.clone()
```

Comment:
Should we return an `Option<&SocketAddr>` instead of cloning (i.e., call `as_ref` instead)?